### PR TITLE
AI Tutor Event

### PR DESCRIPTION
**Pull request by ZiyaSelim**
## Description of changes
Please provide a short description of the change or if the change is sufficiently described in the GitHub issue, provide a link to the issue.

  
## How has this been tested?
Please describe the test strategy you followed.

- [ ] automated unit test
- [ ] automated integration test
- [ ] automated acceptance test
- [x] manual, exploratory test

In case of manual test, please document the test well including a set of user instructions and prerequisites. Each including an action, it's result, and where appropriate a screenshot.
In case of automated tests, the test coverage should be reasonably high, especially on Service classes and other classes with complex logic.
    
## Checklist before requesting a review
- [x] My code follows the [coding guidelines](https://github.com/MEITREX/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My code fulfills all acceptance criteria
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] The test coverage (line and branch) is reasonably high, especially on Service classes and other classes with complex logic
- [x] I have made corresponding changes to the documentation
- [ ] I have added explanation of architectural decision and rationales to [wiki/adr](https://github.com/MEITREX/wiki/tree/main/adr)
- [x] I have updated the changes in the ticket description

## Checklist for reviewer
- [ ] The code works and does not throw errors
- [ ] The code is easy to understand and there are no confusing parts
- [ ] The code follows the [coding guidelines](https://github.com/MEITREX/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] The code change accomplishes what it is supposed to do
- [ ] I cannot think of any use case in which the code does not behave as intended
- [ ] The added and existing tests reasonably cover the code change
- [ ] I cannot think of any test cases, input or edge cases that should be tested in addition
- [ ] Description of the change is included in the documentation

### CODE REVIEW

File: src/main/java/de/unistuttgart/iste/meitrex/common/event/AskedTutorAQuestionEvent.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class AskedTutorAQuestionEvent {
+    private UUID userId;
+    private String question;
+    private TutorCategory category;
+}
```
Comment (myluki2000) [inline src/main/java/de/unistuttgart/iste/meitrex/common/event/AskedTutorAQuestionEvent.java:18]: Da wäre noch wichtig, in welchem Kurs die Frage gestellt wird, wenn es eine kurs-spezifische Frage ist. Die Info tut der Tutor doch glaub auch mit einbeziehen wenn er Fragen beantwortet oder hab ich das falsch im Kopf?

**Review by myluki2000 — CHANGES_REQUESTED** [review]

File: src/main/java/de/unistuttgart/iste/meitrex/common/dapr/TopicPublisher.java [hunk 2]
```diff
@@ -165,6 +165,14 @@ public void notifyStageCompleted(final StageCompletedEvent event) {
         publishEvent(event, DaprTopic.STAGE_COMPLETED);
     }
 
+    /**
+     * Method to notify when a user the AI Tutor a question
+     * @param event of the question that was asked, the corresponding category, the user that asked it and the courseId
+     */
+    public void notifyQuestionAsked(final AskedTutorAQuestionEvent event) {
```
Comment (myluki2000) [inline src/main/java/de/unistuttgart/iste/meitrex/common/dapr/TopicPublisher.java:172]: Würde das hier noch in notifyTutorQuestionAsked() umbennen, nicht dass es mit den Forums-Fragen verwechselt wird

**Review by myluki2000 — CHANGES_REQUESTED** [review]
